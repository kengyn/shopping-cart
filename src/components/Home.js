import React from "react";
import { useNavigate } from "react-router-dom";
const Home = () => {
  let navigate = useNavigate();
  return (
    <>
      <h1>Home</h1>

      <button onClick={() => navigate("shop")}>SHOP NOW</button>
    </>
  );
};

export default Home;
