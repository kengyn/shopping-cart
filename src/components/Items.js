import acquaintedFate from "../assets/Item_Acquaint_Fate.webp";
import intertwinedFate from "../assets/Item_Intertwined_Fate.webp";
import saintQuartz from "../assets/Saintquartz.webp";
import diamonds from "../assets/daimonds.png";
import covenantBookmark from "../assets/Covenant-Bookmark.png";
import galaxyBookmark from "../assets/galaxy-bookmark.png";

const items = [
  {
    id: "acquainted-fate",
    name: "Acquainted Fate",
    price: 1,
    image: acquaintedFate,
    imgLink: "Item_Acquaint_Fate.webp",
  },
  {
    id: "intertwined-fate",
    name: "Intertwined Fate",
    price: 1,
    image: intertwinedFate,
    imgLink: "Item_Intertwined_Fate.webp",
  },
  {
    id: "saint-quartz",
    name: "Saint Quartz",
    price: 1,
    image: saintQuartz,
    imgLink: "Saintquartz.webp",
  },
  {
    id: "diamonds",
    name: "Diamonds",
    price: 1,
    image: diamonds,
    imgLink: "daimonds.png",
  },
  {
    id: "covenant-bookmark",
    name: "Covenant Bookmark",
    price: 1,
    image: covenantBookmark,
    imgLink: "Covenant-Bookmark.png",
  },
  {
    id: "galaxy-bookmark",
    name: "Galaxy Bookmark",
    price: 1,
    image: galaxyBookmark,
    imgLink: "galaxy-bookmark.png",
  },
];

async function getProduct(id) {
  let product = items.find((item) => item.id === id);
  return product ?? null;
}

export { items, getProduct };
