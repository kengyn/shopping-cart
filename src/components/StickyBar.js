import { NavLink, Outlet } from "react-router-dom";
import "../styles/StickyBar.css";

export default function StickyBar() {
  return (
    <>
      <header>
        <div className="nav-header">
          <NavLink to="/">GACHA</NavLink>
        </div>
        <div className="primary-nav">
          <ul>
            <li>
              <NavLink to="shop">Shop</NavLink>
            </li>
            <li>
              <NavLink to="#">Roll Gacha</NavLink>
            </li>
          </ul>
        </div>
        <div className="secondary-nav">
          <div className="search-container">
            <img src={require("../assets/magnify.png")} alt="search" />
            <input type="text" />
          </div>
          <div className="nav-cart">
            <div className="current-cart">0</div>
            <NavLink to="cart">
              <img src={require("../assets/cart-outline.png")} alt="cart" />
            </NavLink>
          </div>
        </div>
      </header>

      <Outlet />
    </>
  );
}
