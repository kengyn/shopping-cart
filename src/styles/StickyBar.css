header {
  display: grid;
  width: 80vw;
  margin: 0 auto;
  grid-template-areas: "logo primary secondary";
  grid-template-columns: 9rem auto 1fr;
  grid-template-rows: 4.75rem minmax(0, auto);
  align-items: center;
}
ul {
  display: flex;
  list-style-type: none;
}

.nav-header {
  grid-area: logo;
  font-size: 2rem;
  font-weight: bold;
}

a {
  text-decoration: none;
  color: black;
  margin-right: 1rem;
}

.primary-nav {
  grid-area: primary;
}

.secondary-nav {
  grid-area: secondary;
  display: flex;
  justify-content: flex-end;
}

.nav-cart {
  display: flex;
  font-size: 1.3rem;
  align-items: center;
}

.search-container {
  position: relative;
  display: flex;
  align-items: center;
  margin-right: 1rem;
}

.search-container input {
  border: 2px solid black;
  background-color: transparent;
  border-radius: 5px;
  height: 2rem;
  font-size: 1rem;
  padding-left: 1rem;
}

.search-container img {
  z-index: -10000;
  height: 1rem;
  position: absolute;
  left: 0.2rem;
}
