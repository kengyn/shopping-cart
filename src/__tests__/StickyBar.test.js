import React from "react";
import { render } from "@testing-library/react";
import { StickyBar } from "../components/StickyBar";
import { MemoryRouter as Router, useLoaderData } from "react-router-dom";

jest.mock("react-router-dom", () => {
  const ogModule = jest.requireActual("react-router-dom");

  return {
    ...ogModule,
    useLoaderData: () => 34,
  };
});

describe("navbar component", () => {
  it("renders 34", () => {
    const result = useLoaderData();
    expect(result).toBe(34);
  });

  it("renders nav bar", () => {
    const { container } = render(
      <Router>
        <StickyBar />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });
});
