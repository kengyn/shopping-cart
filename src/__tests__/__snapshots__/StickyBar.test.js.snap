// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar component renders nav bar 1`] = `
<div>
  <header>
    <div
      class="nav-header"
    >
      <a
        class=""
        href="/shopping-cart"
      >
        GACHA
      </a>
    </div>
    <div
      class="primary-nav"
    >
      <ul>
        <li>
          <a
            class=""
            href="/shop"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            class=""
            href="/gacha"
          >
            Roll Gacha
          </a>
        </li>
      </ul>
    </div>
    <div
      class="secondary-nav"
    >
      <div
        class="search-container"
      >
        <img
          alt="search"
          src="magnify.png"
        />
        <input
          type="text"
        />
      </div>
      <div
        class="nav-cart"
      >
        <div
          class="current-cart"
        >
          34
        </div>
        <a
          class=""
          href="/cart"
        >
          <img
            alt="cart"
            src="cart-outline.png"
          />
        </a>
      </div>
    </div>
  </header>
</div>
`;
